kedro-datasets/kedro_datasets/pandas/sql_dataset.py

```diff
@@ -225,11 +225,12 @@ def __init__(  # noqa: PLR0913
         self.metadata = metadata
 
     @classmethod
-    def create_connection(cls, connection_str: str, connection_args: dict) -> None:
+    def create_connection(cls, connection_str: str, connection_args: dict = None) -> None:
         """Given a connection string, create singleton connection
         to be used across all instances of ``SQLTableDataset`` that
         need to connect to the same source.
         """
+        connection_args = connection_args or {}
         try:
             engine = create_engine(connection_str, **connection_args)
```

Comment:
Note: I do question how/if this `connection_args` should relate to `self._connection_args`, which is defined and used in other locations in that file.

Reply:
Yes it does. `create_connection` gets called within the `engine` property and that's where you can see that for `connection_args` it passes `self._connection_args`: https://github.com/kedro-org/kedro-plugins/blob/main/kedro-datasets/kedro_datasets/pandas/sql_dataset.py#L248

Reply:
Ok. Therefore, what is the purpose of having added a `connection_args` argument in `create_connection()` in the first place?

Shouldn't we simply have called
```
engine = create_engine(connection_str, **self._connection_args)
```
and left `create_connection()` with a single argument (`connection_str`)?

Reply:
With the same argument, connection_str can be replaced with self._connection_str.

I think the main reason for this is connection are shared in `cls.engines`, but not an object attribute. The class property `engine` is using `self._connection_str` and `self._connection_args` already.

Reply:
Would you like me to apply required changes to remove both `connection_str` and `connection_args` arguments from `create_connection()`?